packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -186,15 +186,26 @@ class WidgetSpan extends PlaceholderSpan {
   @override
   bool visitDirectChildren(InlineSpanVisitor visitor) => true;
 
-  @override
+    @override
   InlineSpan? getSpanForPositionVisitor(TextPosition position, Accumulator offset) {
-    if (position.offset == offset.value) {
+    // Record where this WidgetSpan begins in the flat text.
```

Comment:
Good catch, it looks like `WidgetSpan.getSpanForPositionVisitor` did not handle the upstream affinity case.